### ISSUE TITLE
Payments Fraud Detection using DL

### ISSUE THREAD

**Issue by Avdhesh-Varshney** [abhisheks008/DL-Simplified#386]
### Deep Learning Simplified Repository (Proposing new issue)
:red_circle: **Project Title** : Payments fraud detection model</br>
:red_circle: **Aim** : Creating a ML project to detect any fraud in the online payments/transactions to secure the bank and users.</br>
:red_circle: **Dataset** : https://www.kaggle.com/datasets/ealaxi/paysim1</br>
:red_circle: **Approach** : Try to use 3-4 algorithms to implement the models and compare all the algorithms to find out the best fitted algorithm for the model by checking the accuracy scores. Also do not forget to do a exploratory data analysis before creating any model.

**********************************************************************************
### 📍 **Follow the Guidelines to Contribute in the Project :**
* You need to create a separate folder named as the Project Title.
* Inside that folder, there will be four main components.
   * Images - To store the required images.
   * Dataset - To store the dataset or, information/source about the dataset.
   * Model - To store the machine learning model you've created using the dataset.
   * `requirements.txt` - This file will contain the required packages/libraries to run the project in other machines.
* Inside the `Model` folder, the `README.md` file must be filled up properly, with proper visualizations and conclusions.

***********************************************************************
:red_circle::yellow_circle: **Points to Note :**

- The issues will be assigned on a first come first serve basis, 1 Issue == 1 PR.
- "Issue Title" and "PR Title should be the same. Include issue number along with it.
- Follow Contributing Guidelines & Code of Conduct before start Contributing.

***********************************************************************
:white_check_mark: **To be Mentioned while taking the issue :**
- Full name : Avdhesh Varshney
- GitHub Profile Link : https://github.com/Avdhesh-Varshney
- Email ID : avdheshvarshney786@gmail.com
- Participant ID (if applicable): 
- Approach for this Project : 
1. Import necessary libraries.
2. Performing necessary pre-processing.
3. Creating all the required functions.
4. Split data into test and train dataset.
5. Train the model on Decision tree classifier, k nearest, and many test on many other algorithms.
6. Predict the test data and checking the accuracy.
7. At the end, labeled all the accuracy of all models in a table.
- What is your participant role? (Mention the Open Source program) Codepeak'23 contributor.

*************************************************************
Happy Contributing 🚀 

All the best. Enjoy your open source journey ahead. 😎

**Comment by Avdhesh-Varshney** [comment]
@abhisheks008 please assign this project to me to work on it and contribute it under Codepeak'23.

**Comment by abhisheks008** [comment]
Hi @Avdhesh-Varshney it's a nice issue/project to work on. But this project repository mainly focuses on DL projects than ML projects. What are DL methods you wanna use for this project?

**Comment by Dhruv127** [comment]
Can you assign this to me ? 
Will Utilize Long Short-Term Memory Networks (LSTMs) for sequential pattern analysis and Autoencoders for anomaly detection in transactional data to detect fraud in online payments effectively within a Deep Learning framework.

**Comment by abhisheks008** [comment]
Hi @Avdhesh-Varshney are you still willing to work on this issue?

**Comment by Nithish-456** [comment]
Please assign this project to me for contribution under SWOC 2024

**Comment by abhisheks008** [comment]
Hi @Nithish-456 nice to have you here. Can you tell us a brief about the approach you are going to take for solving this issue?

**Comment by Nithish-456** [comment]
Yes sir, My approach is to detect (classify) the fraud transactions in financial banks using 3 or 4 Machine Learning algorithms like random forest classifier, ensemble methods, support vector classifier and so on. So, by employing that algorithms on the provided dataset. I can able to recognize which algorithm is best ... and also we can use feature engineering, feature selection techniques further..

**Comment by abhisheks008** [comment]
As this project repository majorly consists of deep learning projects, I suggest you look out for different DL methods/approaches for this project rather than going for basic ML methods. I am looking forward to hear from you.

**Comment by nilam576** [comment]
I would like to take up this issue
Full name : Nilam Bhojwani
GitHub Profile Link : https://github.com/nilam576
Email ID : nilambhojwaningp@gmail.com
Participant ID (if applicable):
Approach for this Project :
1.Anomaly Detection with Autoencoders
2. Generative Adversarial Networks (GANs) for Adversarial Training:
3.Recurrent Neural Networks (RNNs) for Sequence Patterns

What is your participant role? (Mention the Open Source program) - SWOC'23 contributor.

**Comment by Nithish-456** [comment]
> As this project repository majorly consists of deep learning projects, I suggest you look out for different DL methods/approaches for this project rather than going for basic ML methods. I am looking forward to hear from you.

Sir, in case of DL we can use MLP classifier or building a neural network. But my point of view is there is no sequential data to process, only we can do classification and also deep learning works good for larger datasets... So, if you want to do build a neural network I will go with DL approach sir. Or else I can choose another project issue... I am looking forward to hear suggestion from you. Thanks

**Comment by abhisheks008** [comment]
Sorry of the inconvenience caused by me. Can you find out a dataset which matches with the issue's goal and can work good with neural networks?

**Comment by Madhusri02** [comment]
Full name : Madhu Sri
GitHub Profile Link : https://github.com/Madhusri02
Email ID : madhusri0774@gmail.com
What is your participant role ? : Contributor at SWOC 2024
Approach for this Project : I will use LSTM , CNN , autoencoders to find the tricky patterns and irregularity in transactions.

**Comment by dsid271** [comment]
Full name : Sidhartha Dondapati
GitHub Profile Link : https://github.com/dsid271
Email ID : dsid271@gmail.com
Participant ID (if applicable):
Approach for this Project : 
1. **Dataset Overview:**
   - Utilized the PaySim synthetic dataset, simulating mobile money transactions from an African country.
   - Dataset size: 470 MB, 6,362,620 rows.
   - 11 attributes, target column: 'isFraud'.
   - Attributes: 5 numeric, 4 categorical, 2 string.

2. **Data Preprocessing:**
   - Imbalanced data: 98.87% non-fraud, 0.12% fraud transactions.
   - Applied Synthetic Minority Over Sampling Technique (SMOTE) to balance the dataset, increasing the percent of the minority class.
   - Dropped 'step' attribute due to no correlation with transactions.
   - Removed 'nameOrig' and 'nameDest' (unique values) and 'isFlaggedFraud' (no impact).

3. **Feature Engineering:**
   - Dropped unnecessary attributes for model training.

4. **Modeling Approach:**
   - Utilized a deep learning approach with a Sequential neural network.
   - Applied standardization to numerical features and one-hot encoding to categorical features.
   - Implemented dropout regularization to prevent overfitting.
   - Used the Adam optimizer with a learning rate of 0.001.
   - Trained the model for 10 epochs with a batch size of 64.

5. **Evaluation:**
   - Evaluated the model using accuracy as the performance metric.
   - The model's accuracy was assessed on the test set after training.

What is your participant role? (Mention the Open Source program): SWOC 2024

**Comment by dsid271** [comment]
I will use a simple feed forward neural network

**Comment by dsid271** [comment]
Full name : Sidhartha Dondapati
GitHub Profile Link : https://github.com/dsid271
Email ID : [dsid271@gmail.com](mailto:dsid271@gmail.com)
Participant ID (if applicable):
Approach for this Project : 
I will preprocess the imbalanced data and apply SMOTE, Drop unnecessary attributes for model training, use a feed forward neural network and LSTM and evaluate their accuracy 
What is your participant role? (Mention the Open Source program): SWOC 2024

**Comment by AgrawalTitiksha** [comment]
Full name : Titiksha Agrawal

GitHub Profile Link : https://github.com/AgrawalTitiksha

Email ID : agrawaltn2311@gmail.com

Approach for this Project :
 
1. Import necessary libraries (numpy, matplotlib, seaborn, pandas, tensorflow, keras, openCV)

2. Performing necessary pre-processing (checking for missing values, duplicates)

3. Visualizations for visual understanding of dataset.

4. Creating all the required functions.

5. Split data into test and train dataset.

6. Train the model on Decision tree classifier, k nearest, and many test on many other algorithms. (ML algo, & can try working 
on converting non-image dataset to image and then apply CNN algorithm)

8. Predict the test data and checking the accuracy.

9. At the end, labeled all the accuracy of all models in a table.

What is your participant role? (Mention the Open Source program) : @contributor SWOC

**Comment by abhisheks008** [comment]
Hi @AgrawalTitiksha focus on deep learning methods specially rather than machine learning models, as this project repository demands deep learning projects.

Issue assigned to you @AgrawalTitiksha

**Comment by dsid271** [comment]
Hey @abhisheks008 was my approach not upto your expectations? Can you let me know what I missed?

**Comment by dsid271** [comment]
Noted!
